reflex/.templates/web/utils/state.js

```diff
@@ -312,23 +325,22 @@ export const connect = async (
 
   // Once the socket is open, hydrate the page.
   socket.current.on("connect", () => {
-    setConnectError(null)
+    setConnectErrors([]);
   });
 
-  socket.current.on('connect_error', (error) => {
-    setConnectError(error)
+  socket.current.on("connect_error", (error) => {
+    setConnectErrors((connectErrors) => [...connectErrors, error]);
```

Comment:
i think we should slice out the existing connectErrors to < 10 so we don't balloon the memory of the browser if the backend doesn't come back up for a long period.

reflex retries pretty aggressively and if there's a tab in the background with a disconnected backend (for days) i could see this causing problems.